### PR TITLE
Add OWASP dependency check to build

### PR DESCRIPTION
**Pull request by hepsimo**
- add dependency check to maven and make builds
- bump companies-house-parent POM to 1.3.1
- bump structured-logging to 1.9.14
- fix sonar report path: coverage XML not uploading to sonarqube

Resolves:
BI-13349

### CODE REVIEW

**Review by PhilipGlover — APPROVED** [review]
as agreed in team discussion

File: pom.xml [hunk 1]
```diff
@@ -27,7 +27,7 @@
         <environment-reader-library.version>1.3.2</environment-reader-library.version>
         <!-- Logging -->
         <log4j-bom.version>2.17.0</log4j-bom.version>
-        <structured-logging.version>1.9.12</structured-logging.version>
+        <structured-logging.version>1.9.14</structured-logging.version>
```
Comment (PhilipGlover) [inline pom.xml:30]: should we go back to 1.9.12 for now?

Comment (hepsimo) [reply pom.xml]: I've merged `master` branch in and picked its version, i.e. 1.9.12

Comment (PhilipGlover) [reply pom.xml]: 👍

**Pull request merged**